aws/services/waf/resource.ftl

```diff
@@ -311,3 +311,113 @@
             }
     /]
 [/#macro]
+
+
+[#macro createWAFLoggingFromProfile occurrence wafaclId loggingProfile regional=false ]
+
+    [#local dataFeedArns = []]
+
+    [#if regional ]
+        [#local wafType = "regional" ]
+    [#else]
+        [#local wafType = "global" ]
+    [/#if]
+
+    [#list (loggingProfile.ForwardingRules)!{} as id,forwardingRule ]
+        [#list forwardingRule.Links?values as link]
+            [#if link?is_hash]
+                [#local linkTarget = getLinkTarget(occurrence, link) ]
+
+                [#if !linkTarget?has_content]
+                    [#continue]
+                [/#if]
+
+                [#local linkTargetRoles = linkTarget.State.Roles ]
+
+                [#if (linkTargetRoles.Outbound["logwatcher"]!{})?has_content
+                        && linkTarget.Direction == "outbound" ]
+
+                    [#switch linkTarget.Core.Type ]
+                        [#case DATAFEED_COMPONENT_TYPE ]
+                            [#if linkTarget.Configuration.Solution["aws:WAFLogFeed"] ]
+
+                                [#local dataFeedArns = combineEntities(
+                                        dataFeedArns,
+                                        linkTarget.State.Attributes["STREAM_ARN"],
+                                        UNIQUE_COMBINE_BEHAVIOUR
+                                )]
+
+                            [#else]
+                                [@fatal
+                                    message="Invalid Data Feed Configuration"
+                                    detail="To use datafeed for WAF Logging please enable aws:WAFLogFeed"
+                                    context=linkTarget.Configuration.Solution
+                                /]
+                            [/#if]
+                            [#break]
+                    [/#switch]
+                [/#if]
+            [/#if]
+        [/#list]
+    [/#list]
+
+    [@debug message="WAFLog" context=dataFeedArns enabled=true /]
+
+    [#if dataFeedArns?has_content ]
+        [#if deploymentSubsetRequired("cli", false) ]
```

Comment:
Usually details of subsets are contained in calling setup routines and the service support routines assume they need to do whatever is provided. 

Maybe worth adding the actual subset names as parameters?

Reply:
So are you saying call this macro twice in each component? Once in the setup `cli` subset and once in the setup `epilogue`?

Reply:
So the other approach could be that we have `create` service routines and `setup` routines. Creates are the raw create an output things and `setup` is a more complete functionality which can include multiple subsets

Reply:
The only change I was suggesting was to add the actual subset names as parameters. "cli" and "epilogue" are pretty well wired into the AWS support so I'll merge as is.